ddl/ddl_api.go

```diff
@@ -3132,6 +3202,10 @@ func (d *ddl) DropIndex(ctx sessionctx.Context, ti ast.Ident, indexName model.CI
 	}
 
 	err = d.doDDLJob(ctx, job)
+	// index not exists, but if_exists flags is true, so we ignore this error.
+	if ErrCantDropFieldOrKey.Equal(err) && ifExists {
```

Comment:
report warning here?

Reply:
done in https://github.com/pingcap/tidb/pull/10723/commits/0d609e3ad5b32a09655a369a9cbf8b3bd17ff3b6.